test/custom-worker.js

```diff
@@ -0,0 +1,60 @@
+// This example uses @typescript/vfs to create a virtual TS program
+// which can do work on a bg thread.
+
+// This version of the vfs edits the global scope (in the case of a webworker, this is 'self')
+importScripts("https://unpkg.com/@typescript/vfs@1.3.0/dist/vfs.globals.js")
+
+/** @type { import("@typescript/vfs") } */
+const tsvfs = globalThis.tsvfs
+
+/** @type {import("../src/tsWorker").CustomTSWebWorkerFactory }*/
+const worker = (TypeScriptWorker, ts, libFileMap) => {
+  return class MonacoTSWorker extends TypeScriptWorker {
+
+    // Adds a custom function to the webworker
+    async getDTSEmitForFile(fileName) {
+      const result = await this.getEmitOutput(fileName)
+      const firstDTS = result.outputFiles.find(o => o.name.endsWith(".d.ts"))
+      return (firstDTS && firstDTS.text) || ""
+    }
+
+    async printAST(fileName) {
```

Comment:
@orta this is a very nice example that might come handy to me, thank you for that! I'm wondering though - isn't there a way to just access the existing program? Recreating it like here seems like an overkill for what I need - especially that I'd have to learn how to properly configure it in the very same way that it's done by default and it's easy to do some mistakes in that.

Reply:
This is exactly what I ended up doing:
https://github.com/statelyai/xstate-viz/blob/cd128a15486d9253edaaa360ee6c1b150f293f5c/public/ts-worker.js#L59

but it's not a part of the public interface so feels like a hack. I wonder if this could be added as part of the public API?